core/Repository/NPMRepositoryClient.cs

```diff
@@ -0,0 +1,137 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading.Tasks;
+using Cmf.CLI.Core.Interfaces;
+using Cmf.CLI.Core.Objects;
+using Cmf.CLI.Core.Services;
+using Cmf.CLI.Utilities;
+
+namespace Cmf.CLI.Core.Repository;
+
+public class NPMRepositoryClient : IRepositoryClient
+{
+    private IFileSystem fileSystem;
+    private Uri registryUrl;
+    private INPMClientEx client;
+    public NPMRepositoryClient(string registryUrl, IFileSystem fileSystem = null, INPMClientEx client = null)
+    {
+        this.registryUrl = new Uri(registryUrl);
+        this.client = client ?? new NPMClient(registryUrl);
+        this.fileSystem = fileSystem ?? new FileSystem();
+    }
+    public async Task<CmfPackageV1> Find(string packageId, string version)
+    {
+        try
+        {
+            var pkg = await client.FetchPackageVersion(packageId.ToLowerInvariant(), version);
+            pkg.Client = this;
+            return pkg;
+        }
+        catch (Exception)
+        {
+            return null;
+        }
+    }
+
+    public async Task<CmfPackageV1Collection> List()
+    {
+        // not all feeds support searching for keywords
+        var pkgs = await client.SearchPackages("keywords:cmf-deployment-package");
+        return new CmfPackageV1Collection();
+        // throw new System.NotImplementedException();
+    }
+
+    public async Task Put(CmfPackageV1 package)
+    {
+        var tmp = this.fileSystem.DirectoryInfo.New(this.fileSystem.Path.GetTempPath());
```

Comment:
Since `GetTempPath()` returns the same path every time, and the name of package files are not randomized, don't we run the risk of, if running two `cmf` commands on the same machine, having conflicts of trying to write to the same temporary file?

I think it is unlikely, but still, don't know if creating a random folder inside Temp wouldn't be safer?